src/pretix/presale/checkoutflow.py

```diff
@@ -570,7 +570,7 @@ def question_is_required(q):
                 )
 
             for q in cp.item.questions_to_ask:
-                if question_is_required(q) and not answ.get(q.id):
+                if question_is_required(q) and q.id not in answ.keys():
```

Comment:
```suggestion
                if question_is_required(q) and q.id not in answ:
```

you don't even need a the .keys() for a dict :)

Reply:
I know, but it has been personal preference to be explicit with dict containment.
I'd say its more natural for a dict to iterate over key/value pairs by default, but that's not the way it is :sweat_smile:

Reply:
> I know, but it has been personal preference to be explicit with dict containment.

Nevermind then, it certainly does not hurt to be explicit. Feel free to do it as you like ;)

> I'd say its more natural for a dict to iterate over key/value pairs by default, but that's not the way it is sweat_smile

For iterating, I think I agree. For containment checks, I'd only agree because it would be consistent if iteration, ``key in dict`` does feel natural to me (i.e. you can say it out loud and it's somewhat clear what it means).